### ISSUE TITLE
Incompatible Shapes Error when Training Wavenet

### ISSUE THREAD

**Issue by seatonullberg** [Rayhane-mamah/Tacotron-2#58]
> Exiting due to Exception: Incompatible shapes: [2,12799,10] vs. [2,12799]
	 [[Node: model/loss/sub_7 = Sub[T=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:GPU:0"](model/loss/strided_slice_7, model/loss/Max)]]
	 [[Node: model/loss/truediv/_975 = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/device:CPU:0", send_device="/job:localhost/replica:0/task:0/device:GPU:0", send_device_incarnation=1, tensor_name="edge_20598_model/loss/truediv", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:CPU:0"]()]]

> Caused by op 'model/loss/sub_7', defined at:
  File "train.py", line 127, in <module>
    main()
  File "train.py", line 121, in main
    train(args, log_dir, hparams)
  File "train.py", line 75, in train
    checkpoint = wavenet_train(args, log_dir, hparams, input_path)
  File "/home/seaton/repos/PyPPA/Tacotron-2/wavenet_vocoder/train.py", line 244, in wavenet_train
    return train(log_dir, args, hparams, input_path)
  File "/home/seaton/repos/PyPPA/Tacotron-2/wavenet_vocoder/train.py", line 167, in train
    model, stats = model_train_mode(args, feeder, hparams, global_step)
  File "/home/seaton/repos/PyPPA/Tacotron-2/wavenet_vocoder/train.py", line 118, in model_train_mode
    model.add_loss()
  File "/home/seaton/repos/PyPPA/Tacotron-2/wavenet_vocoder/models/wavenet.py", line 347, in add_loss
    self.loss = DiscretizedMixtureLogisticLoss(self.y_hat[:, :, :-1], self.y[:, 1:, :], hparams=self._hparams, mask=self.mask)
  File "/home/seaton/repos/PyPPA/Tacotron-2/wavenet_vocoder/models/modules.py", line 370, in DiscretizedMixtureLogisticLoss
    log_scale_min=hparams.log_scale_min, reduce=False)
  File "/home/seaton/repos/PyPPA/Tacotron-2/wavenet_vocoder/models/mixture.py", line 70, in discretized_mix_logistic_loss
    log_probs = log_probs + log_prob_from_logits(logit_probs)
  File "/home/seaton/repos/PyPPA/Tacotron-2/wavenet_vocoder/models/mixture.py", line 17, in log_prob_from_logits
    return x - m - tf.log(tf.reduce_sum(tf.exp(x-m), axis))
  File "/home/seaton/.local/lib/python3.6/site-packages/tensorflow/python/ops/math_ops.py", line 894, in binary_op_wrapper
    return func(x, y, name=name)
  File "/home/seaton/.local/lib/python3.6/site-packages/tensorflow/python/ops/gen_math_ops.py", line 4636, in _sub
    "Sub", x=x, y=y, name=name)
  File "/home/seaton/.local/lib/python3.6/site-packages/tensorflow/python/framework/op_def_library.py", line 787, in _apply_op_helper
    op_def=op_def)
  File "/home/seaton/.local/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 2956, in create_op
    op_def=op_def)
  File "/home/seaton/.local/lib/python3.6/site-packages/tensorflow/python/framework/ops.py", line 1470, in __init__
    self._traceback = self._graph._extract_stack()  # pylint: disable=protected-access

This occurred during the training of the Tacotron-2 model just after the Tacotron portion had been completed. Is there any way to get around this?

**Comment by gsoul** [comment]
See the same issue.

**Comment by Rayhane-mamah** [comment]
Hello sorry for the super late answer, I believe I fixed those in latest commit, so I'm closing this.

Feel free to reopen if the problem persists. also, always make sure to use `out_channels = 10 * 3` with "raw and `out_channels = 256` with mulaw-quantize.

**Comment by shartoo** [comment]
here is my configuration in `hparams.py`

```
input_type="raw",
quantize_channels=2 ** 16,  # 65536 (16-bit) (raw) or 256 (8-bit) (mulaw or mulaw-quantize) // number of classes = 256 <=> mu = 255

log_scale_min=float(np.log(1e-14)), #Mixture of logistic distributions minimal log scale
log_scale_min_gauss = float(np.log(1e-7)), #Gaussian distribution minimal allowed log scale

#To use Gaussian distribution as output distribution instead of mixture of logistics, set "out_channels = 2" instead of "out_channels = 10 * 3". (UNDER TEST)
out_channels = 2, #This should be equal to quantize channels when input type is 'mulaw-quantize' else: num_distributions * 3 (prob, mean, log_scale).
layers = 30, #Number of dilated convolutions (Default: Simplified Wavenet of Tacotron-2 paper)
stacks = 3, #Number of dilated convolution stacks (Default: Simplified Wavenet of Tacotron-2 paper)
residual_channels = 512,
gate_channels = 512, #split in 2 in gated convolutions
skip_out_channels = 256,
kernel_size = 3,

cin_channels = 80, #Set this to -1 to disable local conditioning, else it must be equal to num_mels!!
upsample_conditional_features = True, #Whether to repeat conditional features or upsample them (The latter is recommended)
upsample_scales = [15, 20], #prod(upsample_scales) should be equal to hop_size
freq_axis_kernel_size = 3,
leaky_alpha = 0.4,

gin_channels = -1, #Set this to -1 to disable global conditioning, Only used for multi speaker dataset. It defines the depth of the embeddings (Recommended: 16)
use_speaker_embedding = True, #whether to make a speaker embedding
n_speakers = 5, #number of speakers (rows of the embedding)

use_bias = True, #Whether to use bias in convolutional layers of the Wavenet

```
